### ISSUE TITLE
Best way to add columns to an existing Feather file (>10GB) in terms of performance in python

### ISSUE THREAD

**Issue by anmyachev** [apache/arrow#35689]
### Describe the usage question you have. Please include as many useful details as  possible.

The current recommendation is: "Now that Feather supports zero-copy reads into Arrow, the preferred approach would be to zero-copy read the file, add a column, then write a new Feather file" from https://issues.apache.org/jira/browse/ARROW-1518.

This approach is quite costly. I'm very interested to know a more performant way. Thank you.

P.S. I assume that I don't need to write usage details, as this problem is quite common if large data is stored in this format and needs to be updated.

### Component(s)

Other

**Comment by westonpace** [comment]
I don't believe there is a better way today.  The schema is the first thing in the file (so it is not easily modified as extending it will overwrite file contents) and reader currently expects that there is one schema and it applies to all messages in the file.

You could store the extra column in a separate file, and then merge them into a single table in-memory (this merging would be zero-copy).

> P.S. I assume that I don't need to write usage details, as this problem is quite common if large data is stored in this format and needs to be updated.

I believe the most common case is that when a new column is added, all of the existing rows are set to some default value.  E.g. in SQL this would be something like `ALTER TABLE ADD foo DEFAULT 7`.  Commonly this default value is null.

Table formats will often support this sort of thing by noting down the default value in the table metadata.  Then, when scanning existing data, if a column is not present (e.g. this file was uploaded earlier and did not have this column) then the new column is inserted, in memory, at scan time with the default value.

**Comment by thisisnic** [comment]
This issue has been marked as stale because it has had no activity in the past 365 days. Please remove the stale label or comment below, or this issue will be closed in 14 days. If this usage question has evolved into a feature request or docs update, please remove the 'Type: usage' label and add the 'Type: enhancement' label instead.
